pymeasure/adapters/ni_gpib_232ct.py

```diff
@@ -0,0 +1,346 @@
+            return
+        if len(ret_val) < 12:
+            log.warning(f"only {len(ret_val)} bytes received, content: {ret_val}")
+        ret_val = ret_val.lstrip(b"\x00")
+        try:
+            [g_s, g_e, s_e, cnt_raw] = ret_val.splitlines()
+            gpib_stat = self.GPIBStatus(int(g_s))
+            gpib_err = self.GPIBError(int(g_e))
+            ser_err = self.SERIALError(int(s_e))
+            count = int(cnt_raw)
+            log.debug(f"count {count}")
+        except ValueError:
+            g_s = ret_val[: ret_val.find(b"\r")]
+            gpib_stat = self.GPIBStatus(int(g_s))
+            log.warning(f" ACHTUNG! {gpib_stat!a}  \r\n")
+        # Error handling
+        if bool(gpib_stat & self.GPIBStatus.ERR) is True:
+            log.critical(f"Error {self.GPIBError(gpib_err)!a} {self.SERIALError(ser_err)!a}")
+
+    def _read(self, **kwargs):
+        """Read up to (excluding) `read_termination` or the whole read buffer.
+
+        :param kwargs: Keyword arguments for the connection itself.
+        :returns str: ASCII response of the instrument (excluding read_termination).
+        """
+        # log.debug("reading")
+        self.flush_read_buffer()
+        self.connection.write(f"rd #255 {self.address}")
+        time.sleep(0.050)
+        ret_val = self.connection.read()
+        if ret_val != "0":
+            ret_len = self.connection.read()
+            log.debug(f"length of read {ret_len}")
+            self._check_errors()
+        return ret_val
+
+    def _read_bytes(self, count, break_on_termchar, **kwargs):
+        """Read bytes from the instrument.
+
+        :param count: number of bytes to be read. A value of -1 indicates to
+            read from the whole read buffer (defined with the chunk_size).
+        :param bool break_on_termchar: Stop reading at a termination character.
+        :param kwargs: Keyword arguments for the connection itself.
+        :returns bytes: response of the instrument.
+        """
+        # log.debug("read bytes..")
+        if count == -1:
+            count = self.connection.chunk_size - 1
+        self.flush_read_buffer()
+        self.connection.write(f"rd #{count} {self.address}")
+        time.sleep(0.050)
+        ret_val = self.connection.read_bytes(count, break_on_termchar, **kwargs)
+        log.debug(f"read: {ret_val}")
+        time.sleep(0.050)
+        ret_len = self.connection.read()
+        log.debug(f"length of bytes read {ret_len}")
+        self._check_errors()
+        return ret_val
+
+    def _write(self, command, **kwargs):
+        """Write a string to the instrument appending `write_termination`.
+
+        :param str command: Command string to be sent to the instrument
+            (without termination).
+        :param kwargs: Keyword arguments for the connection itself.
+        """
+        self.flush_read_buffer()
+        self.connection.write(f"wrt {self.address} \n  {command}", **kwargs)
```

Comment:
You want to write two spaces between newline and the command itself?

Reply:
Thanks fot point this out,  that was an  oversight..